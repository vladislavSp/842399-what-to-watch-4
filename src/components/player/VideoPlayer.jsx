import React from 'react';
import PropTypes from 'prop-types';
import {withStatePlayer} from './withStatePlayer.jsx';

const VideoPlayer = (props) => {
  const {poster} = props;

  return <video
    poster={ poster }
    autoPlay={ false }
    width="280"
    height="175">
  </video>;
};

VideoPlayer.propTypes = {
  poster: PropTypes.string.isRequired,
};

export default withStatePlayer(VideoPlayer);
