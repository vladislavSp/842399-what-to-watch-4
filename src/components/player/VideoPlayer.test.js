import React from "react";
import renderer from "react-test-renderer";
import VideoPlayer from "./VideoPlayer.jsx";

const isPlaying = false;
const scrVideo = `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`;
const poster = `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`;

describe(`render VideoPlayer`, () => {
  it(`Should VideoPlayer render correctly`, () => {
    const tree = renderer
      .create((
        <VideoPlayer
          isPlaying={isPlaying}
          poster={poster}
          scrVideo={scrVideo}
        />
      ), {
        createNodeMock: () => {
          return {};
        }
      }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
