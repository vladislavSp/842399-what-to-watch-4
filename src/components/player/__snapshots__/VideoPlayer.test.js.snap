// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render VideoPlayer VideoPlayer render correctly 1`] = `
<video
  autoPlay={false}
  height="175"
  poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
  width="280"
/>
`;
